engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java

```diff
@@ -405,29 +405,45 @@ private void validateSortField(SortField sortField, DocumentFieldMetadata fieldM
 			}
 	}
 
-	private void validateNumericSortField(SortField sortField, NumericEncodingType numericEncodingType) {
+	private void validateNumericSortField(SortField sortField, NumericEncodingType indexNumericEncodingType) {
+		final NumericEncodingType sortNumericEncodingType;
 		switch ( sortField.getType() ) {
 			case BYTES:
 			case INT:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.INTEGER );
+				sortNumericEncodingType = NumericEncodingType.INTEGER;
 				break;
 			case LONG:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.LONG );
+				sortNumericEncodingType = NumericEncodingType.LONG;
 				break;
 			case DOUBLE:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.DOUBLE );
+				sortNumericEncodingType = NumericEncodingType.DOUBLE;
 				break;
 			case FLOAT:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.FLOAT );
+				sortNumericEncodingType = NumericEncodingType.FLOAT;
 				break;
 			default:
-				throw LOG.sortTypeDoesNotMatchFieldType( String.valueOf( sortField.getType() ), String.valueOf( numericEncodingType ), sortField.getField() );
+				throw LOG.sortTypeDoesNotMatchFieldType( String.valueOf( sortField.getType() ),
+						String.valueOf( indexNumericEncodingType ), sortField.getField() );
 		}
+		if ( NumericEncodingType.UNKNOWN.equals( indexNumericEncodingType ) ) {
+			/*
+			 * The actual encoding type is unknown, so we can't validate more.
+			 * This happens most notably when using custom numeric field bridges.
+			 * We can't ask the user to add more metadata, since there is no annotation for
```

Comment:
The example I gave above used a different name from the Hibernate property. I see now you meant a different name from the one in the `@Field` annotation.
Just tried exactly that, and it does not work either: https://github.com/yrodiere/hibernate-search/commit/3a15ad43c1d500f1fec3633199a435f985ae0563 and https://travis-ci.org/yrodiere/hibernate-search/builds/153493483
It seems to be a different problem, though. Will investigate.